### PR TITLE
Validate authenticator config before deploying

### PR DESCRIPTION
**Pull request by GeorgianaElena**
Fixes https://github.com/2i2c-org/infrastructure/issues/1125

### CODE REVIEW

File: deployer/config_validation.py [hunk 1]
```diff
@@ -152,3 +152,47 @@ def validate_support_config(cluster_name):
                 sys.exit(1)
     else:
         print_colour(f"No support defined for {cluster_name}. Nothing to validate!")
+
+
+def validate_authenticator_config(cluster_name, hub_name):
+    """
+    Validates the provided authenticator config for each hub of a specific cluster.
```
Comment (choldgraf) [inline deployer/config_validation.py:159]: Could you add a few more comments throughout this function to in plain english what is being checked/validated in each major step?

Comment (choldgraf) [inline deployer/config_validation.py:159]: It seems like this is primarily being used to check whether `auth0` has been disabled if another authenticator is being used, is that right? If so, I think we could make the function description or name more specific. Though if we intend to expand the kind of validation we do in the future then it might be fine.

**Review by choldgraf — COMMENTED** [review]
A few quick comments about adding extra comments etc, but it is looking good to me!

**Comment by GeorgianaElena** [comment]
Thanks @choldgraf! I've added some comments and changed the function name.

**Review by sgibson91 — COMMENTED** [review]
@GeorgianaElena I realise you've already changed the function name once, but I feel this is more of an assert function than validation. What do you think of `assert_single_auth_method_enabled`?

**Comment by GeorgianaElena** [comment]
`assert_single_auth_method_enabled` -> it's just perfect 💯

File: deployer/deploy_actions.py [hunk 2]
```diff
@@ -16,6 +16,7 @@
     validate_cluster_config,
     validate_hub_config,
     validate_support_config,
+    validate_auth0_config,
```
Comment (GeorgianaElena) [inline deployer/deploy_actions.py:19]:
```suggestion
    assert_single_auth_method_enabled,
```

File: deployer/config_validation.py [hunk 3]
```diff
@@ -154,7 +154,7 @@ def validate_support_config(cluster_name):
         print_colour(f"No support defined for {cluster_name}. Nothing to validate!")
 
 
-def validate_auth0_config(cluster_name, hub_name):
+def assert_single_auth_method_enabled(cluster_name, hub_name):
     """
     For each hub of a specific cluster, it validates that when an authenticator
     other than Auth0 is enabled, then `auth0` is explicitly disabled.
```
Comment (sgibson91) [inline deployer/config_validation.py:160]:
```suggestion
    For each hub of a specific cluster, it asserts that only a single auth
    method is enabled. An error is raised when an authenticator
    other than Auth0 is enabled and `auth0` is not explicitly disabled.
```

What do you think?

**Review by sgibson91 — APPROVED** [review]
🚀
